sdkproject/Assets/Mapbox/Unity/Editor/PropertyDrawers/GeometryMaterialOptionsDrawer.cs

```diff
@@ -2,118 +2,135 @@
+			path = Path.Combine(Constants.Path.MAP_FEATURE_STYLES_SAMPLES, Path.Combine(styleName, string.Format("{0}Icon", styleName)));
+		}
+	}
 
 	[CustomPropertyDrawer(typeof(GeometryMaterialOptions))]
 	public class GeometryMaterialOptionsDrawer : PropertyDrawer
 	{
 		static float lineHeight = EditorGUIUtility.singleLineHeight;
-		bool showPosition = true;
+
+		private Dictionary<StyleTypes, StyleIconBundle> _styleIconBundles = new Dictionary<StyleTypes, StyleIconBundle>()
+		{
+			{StyleTypes.Simple, new StyleIconBundle(StyleTypes.Simple.ToString())},
+			{StyleTypes.Realistic, new StyleIconBundle(StyleTypes.Realistic.ToString())},
+			{StyleTypes.Fantasy, new StyleIconBundle(StyleTypes.Fantasy.ToString())},
+			{StyleTypes.Light, new StyleIconBundle(StyleTypes.Light.ToString())},
+			{StyleTypes.Dark, new StyleIconBundle(StyleTypes.Dark.ToString())},
+			{StyleTypes.Satellite, new StyleIconBundle(StyleTypes.Satellite.ToString())},
+		};
+
+		/// <summary>
+		/// Loads the default style icons.
+		/// </summary>
+		private void LoadDefaultStyleIcons()
+		{
+			foreach (var key in _styleIconBundles.Keys)
+			{
+				_styleIconBundles[key].Load();
+			}
+		}
 
 		public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
 		{
+			LoadDefaultStyleIcons();
 			EditorGUI.BeginProperty(position, label, property);
-			//showPosition = EditorGUI.Foldout(new Rect(position.x, position.y, position.width, lineHeight), showPosition, label.text);
-			EditorGUI.LabelField(new Rect(position.x, position.y, position.width, lineHeight), new GUIContent { text = "Material Options", tooltip = "Unity materials to be used for features. " });
+
+			//position.y += lineHeight;
+			var styleTypeLabel = new GUIContent { text = "Texturing Style", tooltip = "Texturing style for feature; choose from sample style or create your own by choosing Custom. " };
+			var styleType = property.FindPropertyRelative("style");
+
+			GUIContent[] styleTypeGuiContent = new GUIContent[styleType.enumDisplayNames.Length];
+			for (int i = 0; i < styleType.enumDisplayNames.Length; i++)
+			{
+				styleTypeGuiContent[i] = new GUIContent
+				{
+					text = styleType.enumDisplayNames[i]
+				};
+			}
+
+			styleType.enumValueIndex = EditorGUILayout.Popup(styleTypeLabel, styleType.enumValueIndex, styleTypeGuiContent);
 			EditorGUI.indentLevel++;
-			//if (showPosition)
+			if ((StyleTypes)styleType.enumValueIndex != StyleTypes.Custom)
+			{
+				GUILayout.BeginHorizontal();
+				var guiStyle = new GUIStyle();
+				guiStyle.imagePosition = ImagePosition.ImageOnly;
+				guiStyle.wordWrap = true;
+				guiStyle.normal.textColor = new Color(0.7f, 0.7f, 0.7f);
+
+				Texture2D thumbnailTexture = (Texture2D)_styleIconBundles[(StyleTypes)styleType.enumValueIndex].texture;
+
+				if (Constants.GUI.StyleLabels.labels.ContainsKey((StyleTypes)styleType.enumValueIndex))
+				{
+					string descriptionLabel = Constants.GUI.StyleLabels.labels[(StyleTypes)styleType.enumValueIndex];
```

Comment:
use `EnumExtensions.Description((StyleTypes)styleType.enumValueIndex)` instead, won't require the if check.